game/songs.cc

```diff
@@ -504,44 +482,45 @@ void Songs::sortChange(int diff) {
-		  case 1: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::collateByTitle)); break;
-		  case 2: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::collateByArtist)); break;
-		  case 3: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::edition)); break;
-		  case 4: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::genre)); break;
-		  case 5: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::path)); break;
-		  case 6: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::language)); break;
+		switch (sortNum(webServer)) {
+		  case 0: std::stable_sort(vector.begin(), vector.end(), customComparator(&Song::randomIdx)); break;
+		  case 1: std::sort(vector.rbegin(), vector.rend(), customComparator(&Song::collateByTitle)); break;
+		  case 2: std::sort(vector.rbegin(), vector.rend(), customComparator(&Song::collateByArtist)); break;
+		  case 3: std::sort(vector.rbegin(), vector.rend(), customComparator(&Song::edition)); break;
+		  case 4: std::sort(vector.rbegin(), vector.rend(), customComparator(&Song::genre)); break;
+		  case 5: std::sort(vector.rbegin(), vector.rend(), customComparator(&Song::path)); break;
+		  case 6: std::sort(vector.rbegin(), vector.rend(), customComparator(&Song::language)); break;
 		  default: throw std::logic_error("Internal error: unknown sort order in Songs::sortChange");
 		}
 	} else {
-		switch (m_order) {
-		  case 0: std::stable_sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::randomIdx)); break;
-		  case 1: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::collateByTitle)); break;
-		  case 2: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::collateByArtist)); break;
-		  case 3: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::edition)); break;
-		  case 4: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::genre)); break;
-		  case 5: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::path)); break;
-		  case 6: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::language)); break;
+		switch (sortNum(webServer)) {
+		  case 0: std::stable_sort(vector.begin(), vector.end(), customComparator(&Song::randomIdx)); break;
+		  case 1: std::sort(vector.begin(), vector.end(), customComparator(&Song::collateByTitle)); break;
+		  case 2: std::sort(vector.begin(), vector.end(), customComparator(&Song::collateByArtist)); break;
+		  case 3: std::sort(vector.begin(), vector.end(), customComparator(&Song::edition)); break;
+		  case 4: std::sort(vector.begin(), vector.end(), customComparator(&Song::genre)); break;
+		  case 5: std::sort(vector.begin(), vector.end(), customComparator(&Song::path)); break;
+		  case 6: std::sort(vector.begin(), vector.end(), customComparator(&Song::language)); break;
```

Comment:
It's the same in both bases except the list is reversed. Isn't there something smart to fix this and just have one switch statement instead of 2?

we could make a list copy which is already in the correct format. and then use it in the switch